subsys/modem/modem_cmux.c

```diff
@@ -584,6 +590,12 @@ static struct modem_cmux_dlci *modem_cmux_find_dlci(struct modem_cmux *cmux)
 
 static void modem_cmux_on_dlci_frame_ua(struct modem_cmux_dlci *dlci)
 {
+	/* Drop invalid UA frames */
+	if (dlci->cmux->frame.cr != dlci->cmux->initiator) {
```

Comment:
Shouldn't this rather be
```suggestion
	if (dlci->cmux->frame.cr == dlci->cmux->initiator) {
```
as we are receiving the frame?